CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -0,0 +1,172 @@
+﻿/*
+ *  Module Name: TexasHoldEmManager.cs
+ *  Purpose: Manages the game of Texas Hold Em.
+ *  Inputs: None
+ *  Outputs: None
+ *  Additional code sources: None
+ *  Developers: Mo Morgan
+ *  Date: 11/32024
+ *  Last Modified: 11/3/2024
+ *  Preconditions: None
+ *  Postconditions: None
+ *  Error/Exception conditions: None
+ *  Side effects: None
+ *  Invariants: None
+ *  Known Faults: None encountered
+ */
+
+using System;
+using System.Collections.Generic;
+using CardsCashCasino.Data;
+
+namespace CardsCashCasino.Manager
+{
+    /// <summary>
+    /// The possible actions that a player can take in Texas Hold Em.
+    /// </summary>
+    public enum PokerAction
+    {
+        FOLD,
+        CHECK,
+        CALL,
+        RAISE,
+        ALL_IN
+    }
+
+    /// <summary>
+    /// The possible roles that a player can have in Texas Hold Em.
+    /// The Small Blind is always left of the Dealer.
+    /// </summary>
+    public enum PlayerRole
+    {
+        DEALER,
+        SMALL_BLIND,
+        BIG_BLIND,
+    }
+    
+    public class TexasHoldEmManager
+    {
+        /// <summary>
+        /// The cardManager for the game.
+        /// </summary>
+        private CardManager _cardManager;
+        
+        /// <summary>
+        /// The user's hand of cards.
+        /// </summary>
+        private UserHand _userHand;
+        
+        /// <summary>
+        /// Flag that indicates if the user is still active in the game.
+        /// </summary>
+        private bool _gameOver;
+        
+        /// <summary>
+        /// The chip manager for the game.
+        /// </summary>
+        private ChipManager _chipManager;
```

Comment:
Like CardManager, lets not initialize this per game. Instead, initialize a single manager in the main game class. The idea is that the Managers communicate with each other through the main game class to function.

This is going to be important when we have cross-game chip inventories.